ORCID depends on an external Maven Repository (which has been unstable)

**Describe the bug**
In our upgrade to Jakarta EE (see #9321) we had to switch to an ORCID API which was compatible with Jakarta EE.  I chose the `org.orcid:orcid-model-jakarta` release from https://artifacts.oicr.on.ca/artifactory/collab-release/

While this was a good quick-fix, this has proven to be less than ideal as the https://artifacts.oicr.on.ca/ site seems to go down for long periods of time (it was down for 3 days earlier this week).  When it is down, anyone attempting to build/recompile DSpace will see build errors like this:
```
[ERROR] Failed to execute goal on project dspace-api: Could not resolve dependencies for project org.dspace:dspace-api:jar:8.0-rc1: Failed to collect dependencies at org.orcid:orcid-model-jakarta:jar:3.3.0: Failed to read artifact descriptor for org.orcid:orcid-model-jakarta:jar:3.3.0: Could not transfer artifact org.orcid:orcid-model-jakarta:pom:3.3.0 from/to oicr.on.ca (https://artifacts.oicr.on.ca/artifactory/collab-release/): Transfer failed for https://artifacts.oicr.on.ca/artifactory/collab-release/org/orcid/orcid-model-jakarta/3.3.0/orcid-model-jakarta-3.3.0.pom 503 Service Unavailable -> [Help 1]
```

Since the official ORCID dependencies (`org.orcid:orcid-model` in Maven Central) have no immediately plans to move to Jakarta EE, we need to find another option.  Two immediate solutions seem possible:
* Reopen https://github.com/DSpace/orcid-jaxb-api and generate our own ORCID API using JAXB with Jakarta EE compatibility.
* Or, fork the codebase behind https://artifacts.oicr.on.ca/artifactory/collab-release/ (https://github.com/dockstore/orcid-model/) and release a copy of it to Maven Central

**Expected behavior**
We need to ensure that all Maven Repositories used by DSpace (in our Parent POM) are trusted & stable.  

Currently, I'm working on reopening our ORCID API again and updating it to work for Jakarta EE:  https://github.com/DSpace/orcid-jaxb-api

--- a/dspace-api/pom.xml
+++ b/dspace-api/pom.xml
@@ -738,26 +738,15 @@
 
         <!-- TODO: This may need to be replaced with the "orcid-model" artifact once this ticket is resolved:
              https://github.com/ORCID/orcid-model/issues/50 -->
+        <!-- Maintained at https://github.com/DSpace/orcid-model -->
         <dependency>
-            <groupId>org.orcid</groupId>
+            <groupId>org.dspace</groupId>
             <artifactId>orcid-model-jakarta</artifactId>
-            <version>3.3.0</version>
+            <version>3.3.0-SNAPSHOT</version>
             <exclusions>
-                <exclusion>
-                    <groupId>com.fasterxml.jackson.jarkarta.rs</groupId>
-                    <artifactId>jackson-jakarta-rs-json-provider</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.yaml</groupId>
-                    <artifactId>snakeyaml</artifactId>
-                </exclusion>
                 <exclusion>
                     <groupId>org.javassist</groupId>
                     <artifactId>javassist</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>io.swagger.core.v3</groupId>
-                    <artifactId>swagger-jaxrs2-jakarta</artifactId>
                 </exclusion>
             </exclusions>
         </dependency>

--- a/dspace-api/pom.xml
+++ b/dspace-api/pom.xml
@@ -742,7 +742,7 @@
         <dependency>
             <groupId>org.dspace</groupId>
             <artifactId>orcid-model-jakarta</artifactId>
-            <version>3.3.0-SNAPSHOT</version>
+            <version>3.3.0</version>
             <exclusions>
                 <exclusion>
                     <groupId>org.javassist</groupId>

--- a/dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/CollectionAdminFeatureIT.java
+++ b/dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/CollectionAdminFeatureIT.java
@@ -175,8 +175,10 @@ public class CollectionAdminFeatureIT extends AbstractControllerIntegrationTest 
         String token = getAuthToken(eperson.getEmail(), password);
 
         // Verify an ePerson in a subgroup of the site administrators has this feature
+        // Filter by "feature" ID because Admins may have >20 features enabled & cause this endpoint to paginate.
         getClient(token).perform(get("/api/authz/authorizations/search/object?embed=feature&uri="
-            + "http://localhost/api/core/sites/" + siteService.findSite(context).getID()))
+            + "http://localhost/api/core/sites/" + siteService.findSite(context).getID()
+            + "&feature=isCollectionAdmin"))
             .andExpect(status().isOk())
             .andExpect(
                 jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='isCollectionAdmin')]")
@@ -284,8 +286,10 @@ public class CollectionAdminFeatureIT extends AbstractControllerIntegrationTest 
         String token = getAuthToken(eperson.getEmail(), password);
 
         // Verify an ePerson in a sub-subgroup of the site administrators has this feature
+        // Filter by "feature" ID because Admins may have >20 features enabled & cause this endpoint to paginate.
         getClient(token).perform(get("/api/authz/authorizations/search/object?embed=feature&uri="
-            + "http://localhost/api/core/sites/" + siteService.findSite(context).getID()))
+            + "http://localhost/api/core/sites/" + siteService.findSite(context).getID()
+            + "&feature=isCollectionAdmin"))
             .andExpect(status().isOk())
             .andExpect(
                 jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='isCollectionAdmin')]")

--- a/pom.xml
+++ b/pom.xml
@@ -102,7 +102,7 @@
                                         <version>${java.version}</version>
                                     </requireJavaVersion>
                                     <requireMavenVersion>
-                                        <version>[3.0.5,)</version>
+                                        <version>[3.8,)</version>
                                     </requireMavenVersion>
                                 </rules>
                             </configuration>
@@ -1990,15 +1990,6 @@
         <repository>
             <id>handle.net</id>
             <url>https://handle.net/maven</url>
-            <snapshots>
-                <enabled>false</enabled>
-            </snapshots>
-        </repository>
-        <!-- For Jakarta EE version of "orcid-model". REMOVE once official "orcid-model" is updated to Jakarta,
-             see https://github.com/ORCID/orcid-model/issues/50-->
-        <repository>
-            <id>oicr.on.ca</id>
-            <url>https://artifacts.oicr.on.ca/artifactory/collab-release/</url>
             <snapshots>
                 <enabled>false</enabled>
             </snapshots>

